pkg/config/config.go

```diff
@@ -150,14 +160,35 @@ func (gc *Config) ShouldSkipTLSForAuthority(candidateAuthority string) bool {
 
 // KappDeployRawOptions returns user configured kapp raw options
 func (gc *Config) KappDeployRawOptions() []string {
-	gc.lock.RLock()
-	defer gc.lock.RUnlock()
+	gc.dataLock.RLock()
+	defer gc.dataLock.RUnlock()
 
 	// Configure kapp to keep only 5 app changes as it seems that
 	// larger number of ConfigMaps negative affects other controllers on the cluster.
 	// Eventually kapp can be smart enough to keep minimal number of app changes.
 	// Set default first so that it can be overridden by user provided options.
-	return append([]string{"--app-changes-max-to-keep=5"}, gc.kappDeployRawOptions...)
+	return append([]string{"--app-changes-max-to-keep=5"}, gc.data.kappDeployRawOptions...)
+}
+
+// AppDefaultSyncPeriod returns duration that is used by Apps
+// that do not explicitly specify sync period.
+func (gc *Config) AppDefaultSyncPeriod() time.Duration {
+	const lowestDefault = 30 * time.Second
+	if gc.data.appDefaultSyncPeriod > lowestDefault {
+		return gc.data.appDefaultSyncPeriod
+	}
+	return lowestDefault
+}
+
+// AppMinimumSyncPeriod returns duration that is used as a lowest
+// sync period App would use for reconciliation. This value
+// takes precedence over any sync period that is lower.
+func (gc *Config) AppMinimumSyncPeriod() time.Duration {
+	const lowestMin = 30 * time.Second
```

Comment:
same question about constants and allocation, especially since for now they're the same value....

Reply:
they are superficially same (same value), but conceptually they are independent.